frontend/src/js/common-ui/forms/CountrySelect.tsx

```diff
@@ -0,0 +1,61 @@
+// Copyright 2024 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+import { Controller } from 'react-hook-form';
+
+import { Autocomplete, TextField } from '@mui/material';
+
+import { countries } from '@northern.tech/store/constants';
+
+interface CountrySelectProps {
+  id?: string;
+  onChange: (...event: any[]) => void;
+  [other: string]: any;
+}
+export const CountrySelect = (props: CountrySelectProps) => {
+  const { id, onChange, ...restProps } = props;
+  return (
+    <Autocomplete
+      getOptionLabel={option => option.label}
```

Comment:
using the `const regionNames = new Intl.DisplayNames(['en'], { type: 'region' });` browser knowledge, you could do 
```suggestion
      getOptionLabel={option => regionNames.of(option)}
```
and reduce maintaining the naming list/ translation